src/RequestHandler/outofprocess/serverprocess.cxx

```diff
@@ -1123,36 +1134,36 @@ SERVER_PROCESS::CheckIfServerIsUp(
 
     if (!g_fNsiApiNotSupported)
     {
-        dwResult = GetExtendedTcpTable(NULL,
-                                       &dwSize,
-                                       FALSE,
-                                       AF_INET,
-                                       TCP_TABLE_OWNER_PID_LISTENER,
-                                       0);
-
-        if (dwResult != NO_ERROR && dwResult != ERROR_INSUFFICIENT_BUFFER)
+        while (dwResult == ERROR_INSUFFICIENT_BUFFER)
         {
-            hr = HRESULT_FROM_WIN32(dwResult);
-            goto Finished;
-        }
+            // Increase the buffer size with additional space, MIB_TCPROW 20 bytes
+            // New entries may be added by other processes before calling GetExtendedTcpTable
+            dwSize += 200;
```

Comment:
make this *2 instead of + 200.

Reply:
double the size is too aggressive. on a regular box with couple IIS sites, the tcptable size is only about 600-800 bytes. Claiming a larger buffer will waste resource.